sympy/core/traversal.py

```diff
@@ -5,6 +5,34 @@
 
 
 
+def iterargs(expr):
+    if isinstance(expr, Basic):
```

Comment:
Why does this check if the arg is Basic? Surely it should only be used with Basic or otherwise it should use sympify.

Also are these functions public? They don't have leading underscore names but they also don't have docstrings.